src/cache-restore.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 The MathWorks, Inc.
+
+import * as cache from '@actions/cache';
+import * as core from '@actions/core';
+import * as crypto from "crypto";
+import { State } from './cache-state';
+import { Release } from './matlab';
+
+export async function restoreMATLAB(release: Release, platform: string, architecture: string, products: string[], matlabPath: string): Promise<boolean> {
+    const installHash = crypto.createHash('sha256').update(products.sort().join('|')).digest('hex')
+    const keyPrefix = `matlab-cache-${platform}-${architecture}-${release.version}`;
+    const primaryKey = `${keyPrefix}-${installHash}`;
+    const cacheKey: string | undefined = await cache.restoreCache([matlabPath], primaryKey);
+
+    core.saveState(State.CachePrimaryKey, primaryKey);
+    core.saveState(State.MatlabCachePath, [matlabPath]);
```

Comment:
Why is this an array rather than a scalar?

Reply:
Changed to scalar because it makes more sense. Originally wanted to check / convert once in cache-save before passing it to cache.saveCache (takes string array). I might've been trying to be "too safe" in the `cache-save` code when getting the value from state.